relay-server/src/actors/spooler/mod.rs

```diff
@@ -673,14 +644,14 @@ impl BufferService {
 
         match self.state {
             BufferState::Memory(ref mut ram)
-            | BufferState::MemoryFileStandby { ref mut ram, .. }
-            // In this state we had full disk, and if we continue spooling without removing
-            // anything from the on-disk spool, we will eventually hit the
-            // `cache.envelope_buffer_size` limit or OOM, whatever comes first.
-            | BufferState::MemoryFileRead { ref mut ram, .. } => {
+            | BufferState::MemoryFileStandby { ref mut ram, .. } => {
                 ram.enqueue(key, managed_envelope);
             }
-            BufferState::Disk(ref disk) =>{
+            BufferState::Disk(ref disk) => {
+                // The disk is full, drop the incoming envelopes.
+                if disk.is_full().await? {
```

Comment:
Idea for a potential follow-up: `is_full` is also called in `transition()` for every `handle_enqueue`, I wonder if we could cache it as a boolean flag on `OnDisk`, which gets updated on dequeue.

Reply:
I'm not sure if I understand -- should `transition()` fail if we try to transition to `OnDisk` when the disk is full?

Reply:
It will never happed, since we do not transition from Disk to Memory if the Disk has some data.